assets/data_table/main.js

```diff
@@ -411,7 +411,7 @@ function App({ ctx, data }) {
         setCurrentMenuForm(currentMenuForm);
       }
     },
-    [filters, content.order_by, content.order]
+    [filters, content.order]
```

Comment:
We send the whole `content` object on every update, so both `content.filters` and `content.order` are different objects (that is, different object references), similarly to how `{} === {}` is actually `false`. `useCallback` and friends only do flat comparison of the dependencies we specify, so the callback will be invalidated whenever we set new `content`, even if the object contents are the same.

So in most cases we don't want to pass objects as dependencies to `use*`, instead specify the individual properties that we care about. For something like filters it's tricky because the number of items may change. One option is to use `JSON.stringify(filters)`, this way we compare the content exactly, but it is a little bit more costly, because we stringify on every render and do string comparison. There's also the option to use `useRef`, such that the callback uses a mutable state.

All that said, does this callback actually depend on filters/order?

Reply:
Hey! In this case I'm using the deps to force the `getCurrentMenuForm` to be reevaluated. What could I do instead? 🤔